### ISSUE TITLE
error: redefinition of ‘class nlohmann::json_pointer’

### ISSUE THREAD

**Issue by FonsDC** [pboettch/json-schema-validator#11]
Hi,
The file json-schema.hpp defines a class `json_pointer`, but a class with the same name is already defined in nlohmann's json.hpp file since version 2.0.0 (2016-06-24). So there is a compilation error. Do we need to use an older version of "JSON for Modern C++"?

**Comment by pboettch** [comment]
No, you need at least version 2.1.0.

I don't have this problem and I think it is related to a `using namespace nlohmann`-clause you might have. Could you confirm this?

If so, please try to do `using nlohmann::json`, if possible for you.

I'm ready to change the name of my class, but I'd first like to understand the problem.

**Comment by FonsDC** [comment]
Where do you suggest to add the `using` -clause?
The error occurs when building the original `json-schema-draft4.json.cpp` from github,  Nothing was changed. See error output below.:
```Scanning dependencies of target json-schema-validator
[ 16%] Building CXX object CMakeFiles/json-schema-validator.dir/src/json-schema-draft4.json.cpp.o
In file included from .../sw/src/git/json-schema-validator/src/json-schema-draft4.json.cpp:1:0:
.../sw/src/git/json-schema-validator/src/json-schema.hpp:64:7: error: redefinition of ‘class nlohmann::json_pointer’
 class json_pointer
       ^
In file included from .../sw/src/git/json-schema-validator/src/json-schema.hpp:39:0,
                 from .../sw/src/git/json-schema-validator/src/json-schema-draft4.json.cpp:1:
.../sw/src/git/nlohmann-json/src/json.hpp:7060:7: error: previous definition of ‘class nlohmann::json_pointer’
 class json_pointer
       ^
``` 
As far as I can see, both `json_pointer` classes are defined in the same namespace `nlohmann`.  Then the error message seems right, and I don't understand why the error does not occur elsewhere  ...

**Comment by FonsDC** [comment]
I did some more tests:
1) Cloned [Modern C++ JSON parser](https://github.com/nlohmann/json) and [Modern C++ JSON schema validator](https://github.com/pboettch/json-schema-validator) on a Windows 10 machine (previous error was on Linux) and build using Visual Studio 14.   It produced the same error:
```
1>  json-schema-draft4.json.cpp
1>D:\sw\src\git\json\json-schema-validator\src\json-schema.hpp(65): error C2011: 'nlohmann::json_pointer': 'class' type redefinition
1>  D:\sw\src\git\json\nlohmann-json\src\json.hpp(6973): note: see declaration of 'nlohmann::json_pointer'

```

2) Renamed `nlohmann::json_pointer` in json-schema.hpp and json-uri.cpp.   No more errors with `json-pointer`, but now compilation failed with an error in json-schema-validate.cpp:
`1>D:\sw\src\git\json\json-schema-validator\app\json-schema-validate.cpp(37): error C2039: 'cerr': is not a member of 'std'
`
This was fixed by adding `#include <iostream>` in json-schema-validate.cpp.
Then the build completed without errors.

**Comment by pboettch** [comment]
OK, I can reproduce this issue with the 'develop'-branch of NLohmann's repository. This branch is currently the default when you clone it. For the moment please use either the master-branch or simply the json.hpp-file from the [releases](https://github.com/nlohmann/json/releases).

**Comment by FonsDC** [comment]
The reason is that in the master-branch, the `json_pointer` class is nested inside `json_base`, so the full name is `nlohmann::json_base::json_pointer`.  But in the development-branch, it has been taken out. 
Thanks for your support!

**Comment by pboettch** [comment]
I will see whether I can get rid of my class in favor of the one from nlohmann.

**Comment by pboettch** [comment]
Fixed with dda72a00238297e3ec1ded268d11c0bb95ff9a85
